src/common/upload.ts

```diff
@@ -106,10 +107,11 @@ export class TurboUnauthenticatedUploadService
     dataItemOpts,
     signal,
     events = {},
-  }: TurboSignedDataItemFactory &
-    TurboAbortSignal &
-    TurboUploadEmitterEvents): Promise<TurboUploadDataItemResponse> {
-    const dataItemSize = dataItemSizeFactory();
+  }: UploadSignedDataItemParams): Promise<TurboUploadDataItemResponse> {
+    const dataItemSize =
+      typeof dataItemSizeFactory === 'number'
+        ? dataItemSizeFactory
+        : dataItemSizeFactory();
```

Comment:
TODO: this typeof number checking was all from debugging. check if we don't need. should be able to just sizeFactory